src/sync/discovery.ts

```diff
@@ -3,57 +3,97 @@ import { getRepositorySummary } from "./jobs";
+		} catch (err) {
+			hasNextPage = false;
+			throw new Error(err);
+		}
+	}
+	return repositories;
+}
+
+// This is a temporary function to assit the feature flag USE_NEW_GITHUB_CLIENT_FOR_DISCOVERY
+// To tidy up, you can replace the call of this function with the true condition block
+const getRepositories = async (app, installationId, jiraHost, logger) => {
+	if (await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_DISCOVERY, true, jiraHost)) {
+		const github = new GitHubClient(getCloudInstallationId(installationId), logger);
+		const repositories = await getAllRepositories(github);
+		// const repositories = await getAllRepositories(github, true);
+		return repositories;
+	}
+
+	const github = await app.auth(installationId);
+	enhanceOctokit(github);
+
+	const repositories = await github.paginate(
+		github.apps.listRepos.endpoint.merge({ per_page: 100 }),
+		(res) => res.data.repositories
+	);
+	return repositories;
+}
 
 export const discovery = (app: Application) => async (job, logger: LoggerWithTarget) => {
 	const startTime = new Date();
 	const { jiraHost, installationId } = job.data;
-	const github = await app.auth(installationId);
-	enhanceOctokit(github);
 
-	try {
-		const repositories = await github.paginate(
-			github.apps.listRepos.endpoint.merge({ per_page: 100 }),
-			(res) => res.data.repositories
-		);
-		logger.info(
-			{ job },
-			`${repositories.length} Repositories found`
-		);
-
-		const subscription = await Subscription.getSingleInstallation(
-			jiraHost,
-			installationId
-		);
-
-		if(!subscription) {
-			logger.info({jiraHost, installationId}, "Subscription has been removed, ignoring job.");
-			return;
-		}
+	const repositories = await getRepositories(app, installationId, jiraHost, logger);
 
-		if (repositories.length === 0) {
-			await subscription.update({
-				syncStatus: SyncStatus.COMPLETE
-			});
-			return;
-		}
+	logger.info(
+		{ job },
+		`${repositories.length} Repositories found`
+	);
 
-		// Store the repository object to prevent doing an additional query in each job
-		// Also, with an object per repository we can calculate which repos are synched or not
-		const repos: Repositories = repositories.reduce((obj, repo) => {
-			obj[repo.id] = { repository: getRepositorySummary(repo) };
-			return obj;
-		}, {});
-		await subscription.updateSyncState({
-			numberOfSyncedRepos: 0,
-			repos
-		});
+	const subscription = await Subscription.getSingleInstallation(
+		jiraHost,
+		installationId
+	);
 
-		await sqsQueues.backfill.sendMessage({installationId, jiraHost, startTime: startTime.toISOString()}, 0, logger);
-	} catch (err) {
-		logger.error({ job, err }, "Discovery error");
+	if(!subscription) {
+		logger.info({jiraHost, installationId}, "Subscription has been removed, ignoring job.");
+		return;
 	}
-};
+
+	if (repositories.length === 0) {
+		await subscription.update({
+			syncStatus: SyncStatus.COMPLETE
+		});
+		return;
+	}
```

Comment:
Can't say I'm a fan of having this duplication of logic for sync.  Can't we reuse the code somehow?